server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala

```diff
@@ -5830,4 +5830,113 @@ trait EmailGetMethodContract {
            |    "header:Subject:asText": "World domination and this is also part of the header"
            |}""".stripMargin)
   }
+
+  @Test
+  def emailGetShouldReturnSpecificHeadersAsAddresses(server: GuiceJamesServer): Unit = {
```

Comment:
What if the mail got this header:

`To: notAMailAddress`

And request `header:To:AsAddress`

?

Reply:
will throw a NPE. I did change the code to return a null value when parsing fails, but end up reverting it because spec says asAddress should be a List. May be an empty list in this case?